### ISSUE TITLE
Selecting all columns from multiple tables generates wrong SQL

### ISSUE THREAD

**Issue by n4tteri** [fsprojects/SQLProvider#477]
### Description

When selecting all columns from multiple tables, generated SQL repeats first table columns for all tables.

### Repro steps
Tested on Oracle 12c with Oracle.ManagedDataAccess (v12.2.1100) and SQLProvider (v1.1.18)

1. Create 2 simple tables:
```sql 
   CREATE TABLE "TABLE_A" 
   (	
      "ID_A" NUMBER GENERATED ALWAYS AS IDENTITY NOT NULL ENABLE,
      "COL_A" NUMBER NOT NULL ENABLE
   );
   
   CREATE TABLE "TABLE_B" 
   (	
      "ID_B" NUMBER GENERATED ALWAYS AS IDENTITY NOT NULL ENABLE,
      "COL_B" NUMBER NOT NULL ENABLE
   );
```

2. Run join or multi-select query with columns from both tables:
```f#
 query { 
    for a in ctx.Main.TableA do
      join b in ctx.Main.TableB on (a.ColA = b.ColB)
      select (a,b) } |> Seq.toArray 
```
Runtime Exception: Oracle.ManagedDataAccess.Client.OracleException: 'ORA-00904: "B"."COL_A": invalid identifier'

Generated SQL:
```sql
SELECT 
   a.COL_A as "a.COL_A",
   a.ID_A  as "a.ID_A",
   b.COL_A as "b.COL_A",
   b.ID_A  as "b.ID_A" 
FROM MAIN.TABLE_A a INNER JOIN  MAIN.TABLE_B b on a.COL_A = b.COL_B
```

### Related information 
* .NET 4.6.2

**Comment by Thorium** [comment]
I tried this with SQLite and it worked correctly. Could this relate to https://github.com/fsprojects/SQLProvider/issues/479

**Comment by n4tteri** [comment]
Tested with the latest version (v1.1.19), unfortunately, the same error.

**Comment by object** [comment]
The generated SQL is definitely wrong, the provider fetches all columns from the first table in the join. I have reproduced it in our database using the latest (1.1.19) release. Will check what's going on.

**Comment by object** [comment]
But if I change the query and explicitly list the columns:
query { 
    for a in ctx.Main.TableA do
      join b in ctx.Main.TableB on (a.ColA = b.ColB)
      select (a.ColA,b.ColB) } |> Seq.toArray 

then everything works fine. So there's something wrong in selecting columns for (select *) case.

**Comment by Thorium** [comment]
There would be some select modifications in PR #484

**Comment by object** [comment]
And when debugging the test example I nailed it down to the following code:

            let selectcolumns =
                if projectionColumns |> Seq.isEmpty then "1" else
                String.Join(",",
                    [|for KeyValue(k,v) in projectionColumns do
                        if v.Count = 0 then   // if no columns exist in the projection then get everything
                            for col in columnCache.[baseTable.FullName] |> Seq.map (fun c -> c.Key) do
                                if singleEntity then yield sprintf "%s.%s as \"%s\"" k col col
                                else yield sprintf "%s.%s as \"%s.%s\"" k col k col
                        else
                            for col in v do
                                if singleEntity then yield sprintf "%s.%s as \"%s\"" k (quoteWhiteSpace col) col
                                else yield sprintf "%s.%s as \"%s.%s\"" k (quoteWhiteSpace col) k col|]) // F# makes this so easy :)

The code after the comment "if no columns exist in the projection then get everything" selects everything but using baseTable.FullName. In JOIN scenario there are several tables involved.

**Comment by object** [comment]
I am not sure if PR #484 addresses this because the incorrect code is inside Oracle provider implementation.

**Comment by object** [comment]
Oops, I see that almost the same code (with the same comment "if no columns exist in the projection then get everything") is duplicated for all providers. Does it means the above scenario fails for every database, not just Oracle?

**Comment by object** [comment]
I must say I don't fully understand the logic here: 
for col in columnCache.[baseTable.FullName] |> Seq.map (fun c -> c.Key) do
    if singleEntity then yield sprintf "%s.%s as \"%s\"" k col col
    else yield sprintf "%s.%s as \"%s.%s\"" k col k col

The check for singleEntity indicates that joins are considered, but the line above assumes all columns belong to baseTable (since it restricts the search to baseTable.FullName). Or am I overlooking somethings?

**Comment by Thorium** [comment]
SqlEntity is basically a dictionary of columnname and value. I'm also not so familiar with that but I expect "if no columns exist in the projection then get everything" is to select the whole entity. The %s thing is a case that if there are same column names in the two tables, we need aliases to the dictionary to know what came from where.

**Comment by Thorium** [comment]
I couldn't create a failing unit-test so I expect others are working fine. Also there are not so many modifications to the code lately, I have updated the versions frequently.

**Comment by Thorium** [comment]
This has been working previously, is it?

**Comment by object** [comment]
I think I know what fails. All code but Oracle looks like this:

for col in columnLookup.[(getTable k).FullName]

Oracle has the following code:

for col in columnCache.[baseTable.FullName]

which is the source of the problem. I will change it and test.

**Comment by object** [comment]
PR is sent.

**Comment by Thorium** [comment]
Merged to 1.1.20
